src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -4162,6 +4161,8 @@ trait Types
       case TypeRef(pre, sym, Nil)                         => copyTypeRef(tycon, pre, sym, args)
       case TypeRef(pre, sym, bogons)                      => devWarning(s"Dropping $bogons from $tycon in appliedType.") ; copyTypeRef(tycon, pre, sym, args)
       case PolyType(tparams, restpe)                      => restpe.instantiateTypeParams(tparams, args)
+      case MethodType(params, resultType)                 => resultType
+      case NullaryMethodType(resultType)                  => resultType
```

Comment:
Does this happen with erroneous trees? Could you add a comment, maybe a `devWarning` like above?